eventsourcing/src/main/java/org/axonframework/eventsourcing/snapshotting/RevisionSnapshotFilter.java

```diff
@@ -135,7 +134,6 @@ public RevisionSnapshotFilter build() {
          */
         protected void validate() {
             assertNonEmpty(type, "The type is a hard requirement and should be provided");
-            assertNonEmpty(revision, "The revision is a hard requirement and should be provided");
```

Comment:
This change actually requires a change in the JavaDoc too, as it states the `revision` is a hard requirement.
However, that's no longer the case, as a `null` revision is perfectly valid.

So, that warrants adjustment on the `Builder` class and the `builder()` method their JavaDoc, removing the fact that `revision is a hard requirement`.
Added, it might be good to state in the documentation that a `null` revision for this filter is fine.